src/lib/stitching/exchange/stitching.ts

```diff
@@ -248,6 +250,82 @@ export const exchangeStitchingEnvironment = ({
             })
           },
         },
+        artworkOrEditionSet: {
+          fragment: gql`
+            ... on CommerceLineItem {
+              artworkId
+              editionSetId
+            }
+          `,
+          resolve: async (parent, _args, context, info) => {
+            const artworkId = parent.artworkId
+            const editionSetId = parent.editionSetId
+
+            if (editionSetId) {
+              return info.mergeInfo.delegateToSchema({
+                schema: localSchema,
+                operation: "query",
+                fieldName: "artwork",
+                args: {
+                  id: artworkId,
+                },
+                context,
+                info,
+                transforms: [
```

Comment:
This is sort of the nutso part--

Before I added the `Artwork.editionSetById` field, there was no way to actually query for an edition set on its own. That meant there was no way for us to delegate a query that could properly resolve a single edition set.

So! I added that field, which essentially does a `find` within an artwork's embedded edition sets to pull out the one you care about.

The next problem-- the whole `delegateToSchema` dance doesn't work out-of-the-box for fields that aren't at the root. I didn't want to expose edition sets at the root (there's not even a way to query for a single edition set in gravity), so I opted for this solution.

The `WrapQuery` takes as an argument the AST version of the nested query, including passing in the `id` argument. This allows us to delegate to the nested field.

Reply:
oh I remember this "not supporting fields that are not in the root" thing!  😱

Reply:
Fascinating! I wonder if there's a way to pass not the full `localSchema` in as the `schema` arg, but just the sub-schema within where `artwork` is at the root (so you preserve using `delegateToSchema` with a root field at the `schema`).

By the way, fetching an edition set by id, sort of makes sense as a root field to me. It's more of an implementation (in Gravity, edition sets are embedded in an artwork), that you'd pretty much never have an edition set 'object' without a parent artwork. That might not be something to necessarily keep enforcing in the graph (ie- why not have an `editionSet` root field?).

Reply:
Whoa interesting thought @mzikherman -- what would that look like?

I think edition sets _could_ be at the root, but I'd rather write an endpoint in gravity to support that first. (and thus far there hasn't really been a need! I don't count this ugliness as a compelling product need just yet 🙈 )